geps/gep-1713/index.md

```diff
@@ -0,0 +1,408 @@
+metadata:
+  name: httproute-example
+spec:
+  parentRefs:
+  - name: child-gateway
+    sectionName: section-in-parent
+```
+
+#### Merging Spec Fields
+
+##### Validation
+
+A parent and child MUST have the same `gatewayClassName`.
+This will be detected by the implementation and reported in [status](#status-fields).
+
+A child resource MUST not set any `spec.infrastructure` fields beyond `attachTo`, and cannot set `spec.address`.
+This can be validated in the CRD schema potentially using CEL.
+
+A parent resource MUST not set `spec.infrastructure.attachTo`.
+That is, we do not allow multiple tiers of Gateways chaining to each other; there is only a single parent with children.
+This can be validated in the CRD schema.
+
+A child resource cannot `attachTo` any Gateway resource that doesn't allow attachment (eg. it does not specify `spec.infrastructure.allowedChildren` for `Gateway`s).
+This will be detected by the implementation and reported in [status](#status-fields).
+
+##### Listeners
+
+Implementations MUST treat the "parent" Gateway as having the concatenated list of all listeners from itself and "child" Gateways.
+
+Validation of this list of listeners MUST behave the same as if the list were part of a single "parent" Gateway.
+
+eg.
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: parent-gateway
+spec:
+  infrastructure:
+    allowedChildren:
+      namespaces:
+        from: Same
+  gatewayClassName: example
+  listeners:
+  - name: HTTP
+    port: 80
+    protocol: HTTP
+---
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: child-gateway
+spec:
+  gatewayClassName: example
+  infrastructure:
+    attachTo:
+      name: parent-gateway
+      kind: Gateway
+      group: gateway.networking.k8s.io
+  listeners:
+  - name: metrics
+    port: 8080
+    protocol: HTTP
+```
+
+With this configuration, the realized "parent" Gateway should listen on port `80` and `8080`.
+
+###### Listener Precedence
+
+Gateway Listeners should be merged using the following precedence:
+- "parent" Gateway
+- "child" Gateway ordered by creation time (oldest first)
+- "child" Gateway ordered alphabetical by “{namespace}/{name}”.
+
+If there are conflicts between these, this should be reported as `Conflicted=True` in the listener as usual.
+
+### Status Fields
+
+#### Addresses
+
+The list of `Addresses` that appear in the status of the "child" Gateway MUST be the same as the "parent" Gateway.
+
+#### AttachedGateways
+
+The property `attachedGateways` will appear on the status of the "parent" Gateway will indicate the total number of "child" Gateways that have are attached
+
+#### Gateway Conditions
+
+Gateway conditions currently supports the following condition types: `Accepted` and `Programmed`
+
+For parent gateways, `Accepted` should be set based on the entire set of merged listeners.
+For instance, if a child listener is invalid, `ListenersNotValid` would be reported.
```

Comment:
Do we already mark Accepted=False for an invalid listener?  If not, then is this only reflecting a "conflicted" status for a listener?  Won't marking Accepted=False make it impossible to attach a xRoute?  I don't think we should make the whole Gateway Accepted=False to account for only one invalid listener.

Reply:
> Do we already mark Accepted=False for an invalid listener? 

It depends on the implementation - it's here https://github.com/kubernetes-sigs/gateway-api/blob/e971a8db22e70b77775622c5ed613c1bb5067e62/apis/v1/gateway_types.go#L825-L836

>  I don't think we should make the whole Gateway Accepted=False to account for only one invalid listener.

The spec allows Accepted=True to be present and to indicate there is a conflict with some listeners.

Details here:
https://github.com/kubernetes-sigs/gateway-api/blob/e971a8db22e70b77775622c5ed613c1bb5067e62/apis/v1/gateway_types.go#L143-L145

I'm going to resolve this since the GEP here is suggesting implementations follow the existing spec.

Reply:
I actually didn't realized we explicitly allowed partially accepting a Gateway as long as Conflicted listeners are excluded already. Agreed we're good with spec alignment here and no further change seems needed.